### ISSUE TITLE
I really beg you resume what it was before the new update! "DataError: AES key data must be 128 or 256 bits"

### ISSUE THREAD

**Issue by Onomatopoesie** [chandler-stimson/live-stream-downloader#15]
It was perfect before the update. What originally could be downlowded perfectly with and only with this add-on, now "DataError: AES key data must be 128 or 256 bits".
Originally this add-on will download the video automatically after I click, and there is just a number on the icon from 1% to 100%, and there will be a ".ts" file on my disk. Perfectly what I need.
Now, with the new update, after I click, there is a new downlowding window (which I think is superfluous), asking me to save as ".mkv", and after that, it may or may not download the video for a few seconds, and then, "DataError: AES key data must be 128 or 256 bits".
I think the version before the new update is so perfect. I really really beg you to resume what it was, or give an independent version of that, otherwise I cannot download what I really need! So much thanks!!!

**Comment by Onomatopoesie** [comment]
I feel so desperate... Please!!!

**Comment by chandler-stimson** [comment]
Please provide an example page

**Comment by Onomatopoesie** [comment]
Thank you for the early reply! Sorry I can't provide the example page because it needs account number and password. May I just give you an example of the m3u8 link (all the videos on the website are in this same kind of type): 
https://livingapihqy.pku.edu.cn/play/harpocrates/2021/09/17/F585CE26BABC447E7D92D229A02039BE/1080_4096040/playlist.m3u8
Both the old version and the new version of the add-on can detect the same ".m3u8" link. After deceting the link, with the old version, I click download button, and the add-on will help me to download it from 1% to 100%, then there will be a ".ts" video file on my disk. With the new version, after I click download button, it first ask me to save as "***.mkv", then it will start downloading, but it whether gives the error "DataError: AES key data must be 128 or 256 bits" immediately, or gives the error a few seconds later after downloading a very small part.
It is so important to me. I am so thankful if there is a way to download it again with your help. Thanks so much.

**Comment by chandler-stimson** [comment]
[test.zip](https://github.com/chandler-stimson/live-stream-downloader/files/7991836/test.zip)

To get keys, it requires credentials. So I cannot test it. Download the "test.zip" version, open "chrome://extensions" in a browser tab. Turn the developer mode on. Disable the old version. Drop this one to the view and try to download the stream. Let me know if it is fixed.

**Comment by chandler-stimson** [comment]
> first ask me to save as "***.mkv",

The extension offers the correct name and file extension when download is over.

**Comment by Onomatopoesie** [comment]
Thank you so much for the help! I've tried with this "test.zip" version as the process you've provided, but the same error exists.

<img width="791" alt="截屏2022-02-03 下午1 03 31" src="https://user-images.githubusercontent.com/98929583/152284258-9cdfcbc8-3f7b-4221-82cf-e6213802c20d.png">

May I kindly ask you provide the old version file before the update and I try again (the version of the below picture, I think the version is 0.1.7)?

<img width="60" alt="截屏2022-02-03 下午1 02 27" src="https://user-images.githubusercontent.com/98929583/152284190-d6ad831d-2cde-40c1-91d0-3046161879b8.png">

**Comment by chandler-stimson** [comment]
You can find the "v2" version on the GitHub. Download the project as ZIP archive and drop the v2 directory to your Chrome the same way you dropped "test.zip"

**Comment by Onomatopoesie** [comment]
Thank you, it works now! I drop the old version of the add-on directory to the web browser, and it can download again as before. Thanks for you early help!!

**Comment by chandler-stimson** [comment]
The "v2" version eventually stops working. You need to provide a sample page that I can play with so that the bug gets resolved on "v3".

**Comment by Onomatopoesie** [comment]
The "v2" version will not work in the future? Then, may I provide you the sample page via email in a while?

**Comment by chandler-stimson** [comment]
That's right. It is based on manifest v2 which gets deprecated by Google. You can send me a temporary login details and I'll investigate `chandler[dot]stimson[]at]gmail.com`

**Comment by Onomatopoesie** [comment]
Email has been sended. Thank you very much!

**Comment by chandler-stimson** [comment]
[live-stream-downloader.zip](https://github.com/chandler-stimson/live-stream-downloader/files/7994196/live-stream-downloader.zip)

I think the bug is fixed. Please give it a try and report back.

**Comment by Onomatopoesie** [comment]
I have tried with this newest version you provide. The problem has been solved, now the videos in question can be downloaded successfully again! The difference is that with the newest version, after the downloading procession is done, the video format is ".mkv", not ".ts" as in the old version downloading from the same source. But both formats can certainly be played without any question.
Deep thanks for your kind help!
